### ISSUE TITLE
Upgrade failed for DaemonSet

### ISSUE THREAD

**Issue by mzac** [traefik/traefik-helm-chart#685]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik-helm-chart/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What version of the Traefik's Helm Chart are you using?

traefik-17.0.2

### What version of Traefik are you using?

2.9.1

### What did you do?

Tried to upgrade my helm DaemonSet deployment

### What did you see instead?

```
Error: UPGRADE FAILED: cannot patch "traefik" with kind DaemonSet: DaemonSet.apps "traefik" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/instance":"traefik-kube-system", "app.kubernetes.io/managed-by":"Helm", "app.kubernetes.io/name":"traefik", "helm.sh/chart":"traefik-17.0.2"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

### What is your environment & configuration?

```bash
helm --debug upgrade --install \
traefik traefik/traefik \
  --namespace kube-system \
  --set deployment.kind=DaemonSet \
  --set service.spec.externalTrafficPolicy=Local \
  --set service.spec.loadBalancerIP=$K3S_TRAEFIK_IP \
  --set ingressClass.isDefaultClass=true \
  --set ports.traefik.expose=true \
  --set service.annotations."metallb\.universe\.tf/address-pool=bgp"
```

**Comment by djbingham** [comment]
I'm seeing a similar error upgrading a non-DaemonSet deployment from 16.0.1 to 17.0.2:

```
Error: UPGRADE FAILED: cannot patch "traefik" with kind Deployment: Deployment.apps "traefik" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/instance":"traefik-traefik", "app.kubernetes.io/managed-by":"Helm", "app.kubernetes.io/name":"traefik", "helm.sh/chart":"traefik-17.0.2"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

Has this been caused by #681, which looks like it copies the `helm.sh/chart` label (along with any others) into selectors? That label is guaranteed to change with every new chart version but, as the error says, the selector field is immutable so this should never work.

**Comment by charlie-haley** [comment]
Good catch @djbingham, we shouldn't be selecting on the labels with a version. I'll raise a patch now to fix this issue

**Comment by charlie-haley** [comment]
The latest release should have a fix for this now, you may still need to recreate the resources to ensure the initial label update is applied correctly. It won't change with every deployment now however.

Let me know if you still have any issues and I can reopen this issue

**Comment by mzac** [comment]
@charlie-haley  After deleting my DaemonSet release and re-installing it worked. Thanks!

**Comment by djbingham** [comment]
@charlie-haley Thank you for the quick response and fix. You are correct, I did have to recreate due to the change of `app.kubernetes.io/instance":"traefik"` to `app.kubernetes.io/instance":"traefik-traefik"` but future upgrades should be fine.

**Comment by trondhindenes** [comment]
this bug is causing is quite a bit of headache since recreating services affect ELBs linked to the traefik service. It would be good if helm charts were tested for "upgradeability" as part of CI to ensure this doesn't happen again.

**Comment by charlie-haley** [comment]
Unfortunately this is unavoidable when making changes to immutable objects and fields @trondhindenes, this is why we increment a major version (in this case, `17.0.0`) to indicate that the version includes breaking changes. We try to avoid making changes like this as much as possible, in this case we had no choice.
We're in the process of improving the chart so the upgrade process is a lot clearer and more accessible to users
